auth/src/main/java/com/firebase/ui/auth/data/model/Resource.java

```diff
@@ -64,20 +80,16 @@ public State getState() {
 
     @Nullable
     public final Exception getException() {
-        mUsed = true;
+        if (mUsable) { mState = State.USED; }
```

Comment:
Yeah, technically, the "resource" part is still immutable since the value and exception can't change. Also, the class is final so only we control the state's mutability, and for now, it can only go from `SUCCESS/FAILURE` -> `USED` so I don't think it'll be a problem unless we want to listen for `USED` events? I'm not sure when that would ever be applicable... I think of the state as more for helping us define our conditionals rather than defining the resource—it should be focused on the data, not the state. That's my opinion though, any digressions?

Reply:
Seems OK for now ... just something we may have to change in the future.